testimony/parser.py

```diff
@@ -51,15 +48,36 @@ def parse(self, docstring=None):
         """
         if docstring is None:
             return {}, {}
+        tokens_dict = {}
         valid_tokens = {}
         invalid_tokens = {}
-        for match in self.token_regex.finditer(docstring):
-            token = match.group(1).strip().lower()
-            value = match.group(2).strip()
+
+        docstring_xml = publish_string(docstring, writer_name='xml')
```

Comment:
This is a good question, got that from previous PR.

From the docstring:

```
    Set up & run a `Publisher` for programmatic use with string I/O.  Return
    the encoded string or Unicode string output.

    For encoded string output, be sure to set the 'output_encoding' setting to
    the desired encoding.  Set it to 'unicode' for unencoded Unicode string
    output.  Here's one way::

        publish_string(..., settings_overrides={'output_encoding': 'unicode'})

    Similarly for Unicode string input (`source`)::

        publish_string(..., settings_overrides={'input_encoding': 'unicode'})

    Parameters: see `publish_programmatically`.
```

It parses the RST from the docstring and return the output as XML